src/sherpa_ai/task_agent.py

```diff
@@ -152,13 +154,19 @@ def run(self, task: str) -> str:
                 logger_step["reply"] = reply_json
             except json.JSONDecodeError:
                 logger_step["reply"] = assistant_reply  # last reply is a string
+            if self.agent_config.verbose:
+                self.verbose_logger.log(f"```{assistant_reply}```")
+
             self.logger.append(logger_step)
 
-            ########## Serial Verbose Feature #######
-            try:
-                formatted_logger_step = show_commands_only(logger_step)
-            except Exception as e:
-                logger.error(e)
+            # Serial Verbose Feature
+            if not self.agent_config.verbose:
+                try:
+                    formatted_logger_step = show_commands_only(logger_step)
+                except Exception as e:
```

Comment:
It's usually not a good idea to silently swallow any type of exception like this. Instead, let it blow up, figure out what specific exception types will be generated, and then add code again to handle each type of exception appropriately.

Reply:
I think in this case we are actually catching KeyError from the function